pyretailscience/analysis/product_association.py

```diff
@@ -226,83 +204,162 @@ def _calc_association(  # noqa: C901 (ignore complexity) - Excluded due to min_*
         if min_uplift < 0.0:
             raise ValueError("Minimum uplift must be greater or equal to 0.")
 
-        unique_combo_df = df[[group_col, value_col]].drop_duplicates()
-        unique_combo_df[value_col] = pd.Categorical(unique_combo_df[value_col], ordered=True)
-        unique_combo_df[group_col] = pd.Categorical(unique_combo_df[group_col], ordered=True)
-
-        sparse_matrix = csc_matrix(
-            (
-                [1] * len(unique_combo_df),
-                (
-                    unique_combo_df[group_col].cat.codes,
-                    unique_combo_df[value_col].cat.codes,
-                ),
-            ),
+        if isinstance(df, pd.DataFrame):
+            df = ibis.memtable(df)
+
+        unique_transactions = df
+        total_transactions = unique_transactions.alias("t")[group_col].nunique().name("total_count")
+
+        product_occurrences = (
+            unique_transactions.group_by(value_col)
+            .aggregate(
+                occurrences=lambda t: t[group_col].nunique(),
+            )
+            .mutate(occurrence_probability=lambda t: t.occurrences / total_transactions)
+            .filter(lambda t: t.occurrences >= min_occurrences)
         )
 
-        row_count = sparse_matrix.shape[0]
+        left_table = unique_transactions.mutate(item_1=unique_transactions[value_col]).drop(value_col)
+        right_table = unique_transactions.mutate(item_2=unique_transactions[value_col]).drop(value_col)
```

Comment:
Would it be better to use rename here? Or does that not work for this case?
```suggestion
        left_table = unique_transactions.rename({"item_1": unique_transactions[value_col]})
        right_table = unique_transactions.rename({"item_2": unique_transactions[value_col]})
```